resilience4j-circuitbreaker/src/main/java/io/github/resilience4j/circuitbreaker/internal/CircuitBreakerStateMachine.java

```diff
@@ -160,48 +161,47 @@ public String toString() {
     public void reset() {
         CircuitBreakerState previousState = stateReference.getAndUpdate(currentState -> new ClosedState(this));
         if (previousState.getState() != CLOSED) {
-            publishStateTransitionEvent(StateTransition.transitionToClosedState(previousState.getState()));
+            publishStateTransitionEvent(StateTransition.transitionBetween(previousState.getState(), CLOSED));
         }
         publishResetEvent();
     }
 
-    @Override
-    public void transitionToClosedState() {
+    private void stateTransition(State newState, Function<CircuitBreakerState, CircuitBreakerState> newStateGenerator) {
         CircuitBreakerState previousState = stateReference.getAndUpdate(currentState -> {
-            if (currentState.getState() == CLOSED) {
+            if (currentState.getState() == newState) {
                 return currentState;
             }
-            return new ClosedState(this, currentState.getMetrics());
+            return newStateGenerator.apply(currentState);
         });
-        if (previousState.getState() != CLOSED) {
-            publishStateTransitionEvent(StateTransition.transitionToClosedState(previousState.getState()));
+        if (previousState.getState() != newState) {
+            publishStateTransitionEvent(StateTransition.transitionBetween(previousState.getState(), newState));
         }
     }
 
+
+    @Override
+    public void disable() {
```

Comment:
I would prefer `transitionToDisabled` over `disable()` and `transitionToForcedOpen` over `forceOpen`. Because it's more consistent to the other methods.